tpchgen/src/random.rs

```diff
@@ -377,16 +380,13 @@ impl RandomPhoneNumber {
         }
     }
 
-    pub fn next_value(&mut self, nation_key: i64) -> String {
-        let country_code = 10 + (nation_key % Self::NATIONS_MAX as i64);
-        let local1 = self.inner.next_int(100, 999);
-        let local2 = self.inner.next_int(100, 999);
-        let local3 = self.inner.next_int(1000, 9999);
-
-        format!(
-            "{:02}-{:03}-{:03}-{:04}",
-            country_code, local1, local2, local3
-        )
+    pub fn next_value(&mut self, nation_key: i64) -> PhoneNumberInstance {
```

Comment:
Returning a structure that can be formatted with `{}`  means the code will (eventually) be able to format the phone number directly into the output stream, rather than formatting (copying) first into an intermediate buffer